test/GPv2Settlement.test.ts

```diff
@@ -186,6 +194,71 @@ describe("GPv2Settlement", () => {
       expect(outTransfers.length).to.equal(tradeCount);
     });
 
+    it("should revert if the limit price is not respected", async () => {
+      const sellAmount = ethers.utils.parseEther("100.0");
+      const sellPrice = 1;
+      const buyAmount = ethers.utils.parseEther("1.0");
+      const buyPrice = 1000;
+
+      const encoder = new SettlementEncoder(testDomain);
+      await encoder.signEncodeTrade(
+        {
+          ...partialOrder,
+          sellAmount,
+          buyAmount,
+          kind: OrderKind.SELL,
+          partiallyFillable: false,
+        },
+        0,
+        traders[0],
+        SigningScheme.TYPED_DATA,
+      );
+
+      expect(toNumberLossy(sellAmount.mul(sellPrice))).to.be.lessThan(
+        toNumberLossy(buyAmount.mul(buyPrice)),
+      );
```

Comment:
Super nitpicky but what about reusing the expression
```
limit_x * price_x >= limit_y * price_y
```
and write it as 
```suggestion
      expect(toNumberLossy(sellAmount.mul(sellPrice))).not.to.be.gte(
        toNumberLossy(buyAmount.mul(buyPrice)),
      );
```
?
([Reference](https://www.chaijs.com/api/bdd/#leastn-msg) to save you a search in case you like the suggestion.)

Reply:
This is a very nice suggestion as it creates symmetry with the actual SC check.